src/data/matrix/basic.lean

```diff
@@ -1370,31 +1370,41 @@ lemma transpose_map {f : α → β} {M : matrix m n α} : Mᵀ.map f = (M.map f)
 by { ext, refl }
 
 /-- `matrix.transpose` as an `add_equiv` -/
+variables (m n α)
 @[simps apply]
 def transpose_add_equiv [has_add α] : matrix m n α ≃+ matrix n m α :=
 { to_fun := transpose,
   inv_fun := transpose,
   left_inv := transpose_transpose,
   right_inv := transpose_transpose,
   map_add' := transpose_add }
-
+variables {m n α}
```

Comment:
I think this should appear after the lemma below

Reply:
And the same for all other similar lemmas?

Reply:
Any equality lemmas for which the types can't be inferred from the other arguments should have these types explicit.

Reply:
Done.